core/src/main/java/bio/terra/pearl/core/service/notification/substitutors/EnrolleeEmailSubstitutor.java

```diff
@@ -80,6 +90,37 @@ public String lookup(String key) {
         return "";
     }
 
+    private record Ternary(String condition, String left, String right) {
+        public static Ternary fromString(String key) {
+            String[] parts = key.split("\\?");
+            String condition = parts[0];
+            String[] values = parts[1].split(":");
+            return new Ternary(condition.trim(), values[0].trim(), values[1].trim());
+        }
+
+        public static boolean isTernary(String key) {
+            return key.contains("?") && key.contains(":") && key.indexOf('?') < key.indexOf(':');
```

Comment:
Might be just my personal preference, but I lean towards (commented) regexes for stuff like this -- something like`/[^:]+\?.+:`